sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala

```diff
@@ -253,7 +253,9 @@ class CacheManager extends Logging with AdaptiveSparkPlanHelper {
    * Tries to re-cache all the cache entries that refer to the given plan.
    */
   def recacheByPlan(spark: SparkSession, plan: LogicalPlan): Unit = {
-    recacheByCondition(spark, _.plan.exists(_.sameResult(plan)))
+    if (spark.sessionState.conf.getConf(SQLConf.QUERY_RESULT_CACHE_AUTO_REFRESH)) {
```

Comment:
I just remember we talked about this: so a better API design is that we do not access the SQLConf as this but have it passed into this method from the call. 

However we have been using this way to get SQLConf so this is fine as it is.

Reply:
BTW I guess in this case the better API design is actually more needed than others.

This is because `cache()` + `SQLConf` becomes the complete API surface for users in this case while maybe a simple `.cache(autoRefresh: Boolean)` makes more sense. 

We could leave this for future exploration.